### PR TITLE
API Room Edit

### PR DESCRIPTION
**Pull request by HugoGresse**
to be reviewed after #285 
- new patch room to edit room
- catch async error which is not possible within an express middleware.

### CODE REVIEW

File: functions/src/api/v1/rooms/createRoom.ts [hunk 1]
```diff
@@ -57,4 +58,4 @@ export const createRoom = async (req: Request, res: Response) => {
         roomId,
         roomSid,
     } as NewRoomResponse)
```
Comment (ntchou) [inline functions/src/api/v1/rooms/createRoom.ts:60]: i think it's not usefull to precise type

Comment (HugoGresse) [reply functions/src/api/v1/rooms/createRoom.ts:60]: I remember why, it's to clarify for the client part which interface/response to expect.

Comment (ntchou) [reply functions/src/api/v1/rooms/createRoom.ts:60]: ok

maybe would be better ?
```
async (req: Request, res: Response):Promise<NewRoomResponse> => {
...
 res.send({
        roomId,
        roomSid,
})
}
``

Comment (HugoGresse) [reply functions/src/api/v1/rooms/createRoom.ts:60]: I can't as I'm not returning but calling res.send function, so probably extend or type the Response if there is a geneneric

Comment (ntchou) [reply functions/src/api/v1/rooms/createRoom.ts:60]: You're right. I review on my phone :)

**Pull request merged**